AppController/djinn.rb

```diff
@@ -2962,6 +2960,11 @@ def restore_appcontroller_state()
     Djinn.log_info("Reload State : #{json_state}")
 
     @@secret = json_state['@@secret']
+    # Is the state we are reading current?
+    if @@secret != HelperFunctions.get_secret()
```

Comment:
It should be, since the secret is passed around by the headnode. If
this is part of the old deployment, then it doesn't matter, since the
state should still be valid. This test should fail only if the state
belongs to a different deployment: the node should then wait for the
headnode.


On Sat, 2016-07-23 at 19:35 -0700, Meni Vaitsi wrote:
> In AppController/djinn.rb:
> > @@ -2962,6 +2960,11 @@ def restore_appcontroller_state()
> >      Djinn.log_info("Reload State : #{json_state}")
> >  
> >      @@secret = json_state['@@secret']
> > +    # Is the state we are reading current?
> > +    if @@secret != HelperFunctions.get_secret()
> Won't this condition be satisfied if I restart my AppScale
> deployment?
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.